src/store/plugins/walletsPlugin.js

```diff
@@ -411,6 +419,47 @@ class WalletsPlugin {
       isSimplified ? { walletId } : {},
     )
   }
+
+  requestETHBalanceByAddress = async (address: Address): Promise<string> => {
+    const network: Network = this.getNetwork()
+
+    try {
+      const ethBalance: string = await web3.getAssetBalance(network, address, 'Ethereum')
+
+      return new BigNumber(ethBalance)
+    } catch (error) {
+      return Promise.delay(MINUTE).then(() => this.requestETHBalanceByAddress(address))
+    }
+  }
+
+  requestETHBalanceByXPUB = async (
+    walletId: WalletId,
+    derivationIndex: number,
+  ): Promise<BigNumber> => {
+    const addresses: Address[] = this.getAddresses(walletId, 0, derivationIndex)
+
+    return Promise
```

Comment:
Promise.map is not standard promise API
Can be replaced by
```
const foo = addresses.map(address => this.requestETHBalanceByAddres(address))
Primise.all(foo).reduce(...
```

Not need to fix now, if works good